library/src/test/java/org/apache/apex/malhar/lib/state/managed/SliceBloomFilterTest.java

```diff
@@ -0,0 +1,191 @@
+        bloomFilter.put(buffer.toSlice());
+      }
+    }
+    buffer.getWindowedBlockStream().releaseAllFreeMemory();
+
+    int falsePositive = 0;
+    for (int i = 0; i < loop; i++) {
+      buffer.writeInt(i);
+      if (!bloomFilter.mightContain(buffer.toSlice())) {
+        Assert.assertTrue(i % span != 0);
+      } else {
+        // BF says its present
+        if (i % 2 != 0) {
+          // But was not there
+          falsePositive++;
+        }
+      }
+    }
+    buffer.getWindowedBlockStream().releaseAllFreeMemory();
+    // Verify false positive prob
+    double falsePositiveProb = falsePositive;
+    falsePositiveProb /= loop;
+    Assert.assertTrue(falsePositiveProb <= expectedFalseProbability);
+
+    for (int i = 0; i < loop; i++) {
+      if (i % span == 0) {
+        buffer.writeInt(i);
+        Assert.assertTrue(bloomFilter.mightContain(buffer.toSlice()));
+      }
+    }
+    buffer.getWindowedBlockStream().releaseAllFreeMemory();
+  }
+
+  private static class FilterOperator extends BaseOperator
+  {
+    private SliceBloomFilter bloomFilter = new SliceBloomFilter(10000, 0.99);
+    private SerializationBuffer buffer = SerializationBuffer.READ_BUFFER;
+
+    public final transient DefaultInputPort<String> input = new DefaultInputPort<String>()
+    {
+      @Override
+      public void process(String tuple)
+      {
+        processTuple(tuple);
+      }
+    };
+
+    @Override
+    public void setup(Context.OperatorContext context)
+    {
+    }
+
+    private int count = 0;
+
+    public void processTuple(String tuple)
+    {
+      buffer.writeString(tuple);
+      bloomFilter.mightContain(buffer.toSlice());
+      buffer.reset();
+    }
+  }
+
+  private static class TestInputOperator extends BaseOperator implements InputOperator
+  {
+    public final transient DefaultOutputPort<String> data = new DefaultOutputPort<String>();
+    private int current = 0;
+
+    @Override
+    public void emitTuples()
+    {
+      data.emit("" + current++);
+    }
+  }
+
+  /**
+   * Just test SliceBloomFilter can be used by operator. such as it is serializable etc
+   * @throws Exception
+   */
+  @Test
+  public void testBloomFilterForApplication() throws Exception
+  {
+    Configuration conf = new Configuration(false);
+
+    LocalMode lma = LocalMode.newInstance();
+    DAG dag = lma.getDAG();
+
+    TestInputOperator generator = new TestInputOperator();
+    dag.addOperator("Generator", generator);
+
+    FilterOperator filterOperator = new FilterOperator();
+    dag.addOperator("filterOperator", filterOperator);
+    dag.addStream("Data", generator.data, filterOperator.input).setLocality(Locality.CONTAINER_LOCAL);
+
+    StreamingApplication app = new StreamingApplication()
+    {
+      @Override
+      public void populateDAG(DAG dag, Configuration conf)
+      {
+      }
+    };
+
+    lma.prepareDAG(app, conf);
+
+    // Create local cluster
+    final LocalMode.Controller lc = lma.getController();
+    lc.run(3000);
```

Comment:
@bhupeshchawda can you please look at this test and remove the timing dependency?